src/solvers/refinement/string_constraint_generator_valueof.cpp

```diff
@@ -437,12 +439,14 @@ exprt string_constraint_generatort::add_axioms_for_parse_int(
     not_exprt(or_exprt(starts_with_minus, starts_with_plus));
 
   /// TODO: we should throw an exception when this does not hold:
-  exprt correct=add_axioms_for_correct_number_format(str, radix);
+  std::size_t max_string_length=40;
+  exprt correct=add_axioms_for_correct_number_format(
+    str, radix, max_string_length);
   axioms.push_back(correct);
 
   /// TODO(OJones): size should depend on the radix
   /// TODO(OJones): we should deal with overflow properly
-  for(unsigned size=1; size<=10; size++)
+  for(unsigned size=1; size<=max_string_length; size++)
```

Comment:
As per our discussion, I will leave this detail because we are rewriting this function in diffblue/test-gen#702